### PR TITLE
AnnularMeshGenerator apply offset in boundary ids and prefix on boundary names

### LINKED ISSUE #25167: Add boundary id offset and name prefix to AnnularMeshGenerator
## Reason
The ability to apply an offset in boundary ids and prefix to boundary names is very useful when we are using multiple mesh generators. This avoid confusion before combine different generated meshes without having extra manipulation on the ids and names.

## Design
following how `GeneratedMeshGenerator` realizes the same ability and make the same function in `AnnularMeshGenerator`

## Impact
`AnnularMeshGenerator` can apply offset on its boundary ids and prefix on its boundary names. We do not need `RenameBoundary` before combining multiple generated meshes.

### PR DESCRIPTION
**Pull request by BoZeng1997**
allow AnnularMeshGenerator to apply offset in boundary ids and prefix on boundary names

closes #25167

### CODE REVIEW

File: framework/include/meshgenerators/AnnularMeshGenerator.h [hunk 1]
```diff
@@ -66,4 +66,10 @@ class AnnularMeshGenerator : public MeshGenerator
 
   /// Whether to construct rings to have equal areas
   const bool & _equal_area;
+
+  /// prefix string for the boundary names
+  const std::string _boundary_name_prefix;
```
Comment (GiudGiud) [inline framework/include/meshgenerators/AnnularMeshGenerator.h:71]:
```suggestion
  const BoundaryName _boundary_name_prefix;
```

File: framework/src/meshgenerators/AnnularMeshGenerator.C [hunk 2]
```diff
@@ -71,6 +71,10 @@ AnnularMeshGenerator::validParams()
                              "dmax!=360, a sector of an annulus or disc is created. In this case "
                              "boundary sidesets are also created at dmin and dmax, and "
                              "given these names");
+  params.addParam<std::string>("boundary_name_prefix",
```
Comment (GiudGiud) [inline framework/src/meshgenerators/AnnularMeshGenerator.C:74]:
```suggestion
  params.addParam<BoundaryName>("boundary_name_prefix",
```

File: framework/src/meshgenerators/AnnularMeshGenerator.C [hunk 3]
```diff
@@ -71,6 +71,10 @@ AnnularMeshGenerator::validParams()
...
+                               "If provided, prefix the built in boundary names with this string");
```
Comment (GiudGiud) [inline framework/src/meshgenerators/AnnularMeshGenerator.C:75]:
```suggestion
                                "If provided, prefix the built in boundary names with this string");
```

File: framework/src/meshgenerators/AnnularMeshGenerator.C [hunk 4]
```diff
@@ -96,7 +100,11 @@ AnnularMeshGenerator::AnnularMeshGenerator(const InputParameters & parameters)
     _full_annulus(_dmin == 0.0 && _dmax == 360),
     _quad_subdomain_id(getParam<SubdomainID>("quad_subdomain_id")),
     _tri_subdomain_id(getParam<SubdomainID>("tri_subdomain_id")),
-    _equal_area(getParam<bool>("equal_area"))
+    _equal_area(getParam<bool>("equal_area")),
+    _boundary_name_prefix(isParamValid("boundary_name_prefix")
+                              ? getParam<std::string>("boundary_name_prefix") + "_"
```
Comment (GiudGiud) [inline framework/src/meshgenerators/AnnularMeshGenerator.C:105]:
```suggestion
                              ? getParam<BoundaryName>("boundary_name_prefix") + "_"
```

File: framework/src/meshgenerators/AnnularMeshGenerator.C [hunk 5]
```diff
@@ -292,7 +300,22 @@ AnnularMeshGenerator::generate()
...
-  mesh->prepare_for_use();
+  // apply boundary id offset and name prefix
+  const auto mesh_boundary_ids = boundary_info.get_boundary_ids();
+  for (auto rit = mesh_boundary_ids.rbegin(); rit != mesh_boundary_ids.rend(); ++rit)
+  {
+
+    const std::string old_sideset_name = boundary_info.sideset_name(*rit);
+    const std::string old_nodeset_name = boundary_info.nodeset_name(*rit);
+
+    MeshTools::Modification::change_boundary_id(*mesh, *rit, *rit + _boundary_id_offset);
```
Comment (GiudGiud) [inline framework/src/meshgenerators/AnnularMeshGenerator.C:311]:
```suggestion
    if (_boundary_id_offset != 0)
      MeshTools::Modification::change_boundary_id(*mesh, *rit, *rit + _boundary_id_offset);
```

File: framework/src/meshgenerators/AnnularMeshGenerator.C [hunk 6]
```diff
@@ -292,7 +300,22 @@ AnnularMeshGenerator::generate()
...
 
+    boundary_info.sideset_name(*rit + _boundary_id_offset) =
```
Comment (GiudGiud) [inline framework/src/meshgenerators/AnnularMeshGenerator.C:313]: same with .empty() on the string prefix

File: test/tests/meshgenerators/annular_mesh_generator/tests [hunk 7]
```diff
@@ -76,4 +76,15 @@
...
     issues = '#20422'
   []
+  [boundary_prefix_offset]
+    type = Exodiff
+    input = boundary_prefix_offset.i
+    cli_args = '--mesh-only'
+    exodiff = boundary_prefix_offset_in.e
+    mesh_mode = 'REPLICATED'
+    recover = false
+    requirement = 'The system shall have the capability of generating an annular mesh with user defined prefix on boundary name and offset on boundary ids'
```
Comment (GiudGiud) [inline test/tests/meshgenerators/annular_mesh_generator/tests:86]:
```suggestion
    requirement = 'The system shall have the capability of generating an annular mesh with a user-defined prefix on boundary names and/or an offset on the boundary ids'
```

**Review by GiudGiud — CHANGES_REQUESTED** [review]
Seems we re baking the same renaming/offseting etc operations everywhere.

we need to think of a more general interface some day

File: framework/src/meshgenerators/AnnularMeshGenerator.C (hunk 3 above)
Comment (BoZeng1997) [reply framework/src/meshgenerators/AnnularMeshGenerator.C:75]: is this a clang format suggestion?

File: framework/src/meshgenerators/AnnularMeshGenerator.C (hunk 6 above)
Comment (BoZeng1997) [reply framework/src/meshgenerators/AnnularMeshGenerator.C:313]: i do not understand your suggestion.

Comment (GiudGiud) [reply framework/src/meshgenerators/AnnularMeshGenerator.C:313]: Don’t change the name of the user did not pass a prefix

File: framework/src/meshgenerators/AnnularMeshGenerator.C (hunk 5 above)
Comment (GiudGiud) [reply framework/src/meshgenerators/AnnularMeshGenerator.C]: reminder.
dont trigger this if there is an offset of 0

File: framework/src/meshgenerators/AnnularMeshGenerator.C (hunk 3 above)
Comment (GiudGiud) [reply framework/src/meshgenerators/AnnularMeshGenerator.C]: yeah so it does not fail clang-format

File: framework/src/meshgenerators/AnnularMeshGenerator.C (hunk 6 above)
Comment (GiudGiud) [reply framework/src/meshgenerators/AnnularMeshGenerator.C]: Added it

File: framework/src/meshgenerators/AnnularMeshGenerator.C (hunk 5 above)
Comment (GiudGiud) [reply framework/src/meshgenerators/AnnularMeshGenerator.C]: added it

**Review by GiudGiud — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**